### PR TITLE
Prefer stateless function site wide

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

This is in preparation to turn back on the [prefer-stateless-function](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md) ESLint rule

## Screenshots

With the plugin set to errors:

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx [hunk 1]
```diff
@@ -6,44 +6,42 @@ export const VAMC_PATHS = {
   PITTSBURGH: /^\/pittsburgh-health-care\/(.)*$/,
 };
 
-export default class VAMCWelcomeModal extends React.Component {
-  static propTypes = {
-    announcement: PropTypes.shape({
-      region: PropTypes.string.isRequired,
-    }).isRequired,
-  };
+function VAMCWelcomeModal({ dismiss, announcement }) {
```
Comment (jhonnyoddball) [inline src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx:9]: isn't it better to add the `export default` here or fix the next file to add it at the bottom for consistency

**Comment by jhonnyoddball** [comment]
Very nice job on this

File: src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx (hunk 1 above)
Comment (bkjohnson) [reply src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx:9]: This one has `propTypes`, so in order to preserve that I think I have to add them after the function definition.  I kinda assumed that once a function had been exported you couldn't add propTypes, but I could be wrong.

File: src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx [hunk 2]
```diff
@@ -6,44 +6,42 @@ export const VAMC_PATHS = {
...
+  );
+}
 
-  render() {
-    const { dismiss } = this.props;
+VAMCWelcomeModal.propTypes = {
+  announcement: PropTypes.shape({
+    region: PropTypes.string.isRequired,
+  }).isRequired,
+};
```
Comment (ncksllvn) [inline src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx:45]: I liked the `static` keyword 😞

**Review by ncksllvn — APPROVED** [review]

File: src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx (hunk 2 above)
Comment (bkjohnson) [reply src/platform/site-wide/announcements/components/VAMCWelcomeModal.jsx:45]: Heartbreak comes in many forms.

**Pull request merged**